rclcpp/test/rclcpp/executors/test_executors.cpp

```diff
@@ -869,3 +856,75 @@ TEST(TestExecutors, testSpinWithNonDefaultContext)
 
   rclcpp::shutdown(non_default_context);
 }
+
+template<typename T>
+class TestBusyWaiting : public ::testing::Test
+{
+public:
+  void SetUp()
+  {
+    rclcpp::init(0, nullptr);
+
+    executor = std::make_shared<T>();
+
+    const auto test_info = ::testing::UnitTest::GetInstance()->current_test_info();
+    std::stringstream test_name;
+    test_name << test_info->test_case_name() << "_" << test_info->name();
+    node = std::make_shared<rclcpp::Node>("node", test_name.str());
+
+    auto waitable_interfaces = node->get_node_waitables_interface();
+    waitable = std::make_shared<TestWaitable>();
+    waitable_interfaces->add_waitable(waitable, nullptr);
+    executor->add_node(node);
+
+    // spin a bit to get all events from the addition of the node processed
+    executor->spin_all(std::chrono::milliseconds(100));
+
+    start_is_ready_count = waitable->get_is_ready_call_count();
+
+    start_time = std::chrono::system_clock::now();
+  }
+
+  void TearDown()
+  {
+    // there were ne events ready, a test iteration should come back in less than 100 ms
+    EXPECT_LT(
+      std::chrono::system_clock::now() - start_time,
+      max_duration / 2)
+      << "spin_some() took a long time to execute when it should have done "
```

Comment:
nitpick: test includes `spin_all` as well.

```suggestion
      << "executor took a long time to execute when it should have done "
```